src/ucp/core/ucp_worker.c

```diff
@@ -2949,6 +2949,10 @@ void ucp_worker_discard_uct_ep(ucp_ep_h ucp_ep, uct_ep_h uct_ep,
     ucs_assert(uct_ep != NULL);
     ucs_assert(purge_cb != NULL);
 
+    if (ucp_is_uct_ep_failed(uct_ep)) {
+        return;
+    }
```

Comment:
IMO better to move this check to `ucp_worker_discard_tl_uct_ep`, it could be clearer why needed

Reply:
fixed